Source/WebCore/animation/DocumentTimeline.cpp

```diff
@@ -299,10 +299,8 @@ void DocumentTimeline::removeReplacedAnimations()
         //    Otherwise, queue a task to dispatch removeEvent at animation. The task source for this task is the DOM manipulation task source.
         auto scheduledTime = [&]() -> std::optional<Seconds> {
             if (auto* documentTimeline = dynamicDowncast<DocumentTimeline>(animation->timeline())) {
-                if (auto currentTime = documentTimeline->currentTime()) {
-                    ASSERT(currentTime->time());
-                    return documentTimeline->convertTimelineTimeToOriginRelativeTime(*currentTime->time());
-                }
+                auto currentTime = MonotonicTime::now().secondsSinceEpoch();
```

Comment:
Shouldn't this use some snapshotted time, like the time for the current event loop update?

Reply:
Here's what I put in the changeling about this change:

> Use the live current time instead of the cached current time so that we can guarantee that "remove" events are dispatched after "finish" events for the same animation.

This scheduling time is not exposed to the Web and is used within `compareStyleOriginatedAnimationEvents()` for event sorting.